### PR TITLE
Tutanarch

### PR DESCRIPTION
**Pull request by darioizzo**
* Tutorials added on the basic classes arch and island.
 * Change in the island streaming operator
 * Change in the pygmo module import (islands now under islands namespace too)

### CODE REVIEW

File: doc/sphinx/credits.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+.. _credits:
+
+Credits 
+============
+
+Who is behind the mysterious and ubiquitous "pagmo development team"?
+What is the elusive and yet so resourceful "pagmo foundation"? Who pays for all of this? Who is using it?
+
+Pagmo/pygmo is a software developed and offered to the users in a full FLOSS phylosophy. We have
```
Comment (bluescarni) [inline doc/sphinx/credits.rst]: philosophy

File: doc/sphinx/credits.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+trajectories as well as designs for spacecarft parts and more, but its capabilities go well beyond these
+applications.
+
+Over the years, our work attracted a wider interest. We acknowledge and thanks the support of
+Google and The Dow Corporation during some phases of the development, and of the European Space Agency
+for having put us together in the Advanced Concepts Team, back in 2006, where it all started.
+
+Main developers
+^^^^^^^^^^^^^^^
+Francesco Biscani - Max Plank Institute for Astronomy
```
Comment (bluescarni) [inline doc/sphinx/credits.rst]: Planck

File: doc/sphinx/docs/python/tutorials/coding_udi.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+the :class:`~pygmo.population` ``pop``.
+
+.. doctest::
+
+    >>> import pygmo as pg
+    >>> class my_isl:
+    ...     def run_evolve(self, algo, pop):
+    ...         return algo.evolve(pop)
+    ...     def get_name(self):
+    ...         return "Its my island!"
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udi.rst]: It's

File: doc/sphinx/docs/python/tutorials/coding_udi.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+   ...         mp_island.init_pool()
+   ...
+   ...     def run_evolve(self, algo, pop):
+   ...         with mp_island._pool_lock:
+   ...             res = mp_island._pool.apply_async(_evolve_func, (algo, pop))
+   ...         return res.get()
+
+The full details are here not reported and can be read in the :class:`~pygmo.mp_island` code. In a nutshell, what happens is that the ``algo.evolve(pop)`` gets offloaded to
+a process (in a shared pool inited upon construction calling the ``init_pool`` static method). The instruction ``res.get()``, makes the thread where ``run_evolve``
+remain waiting for the process execution and while doing so it releases the GIL, making parallelizaion effective.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udi.rst]: parallelization

File: doc/sphinx/docs/python/tutorials/coding_udi.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+   ...         # Init the process pool, if necessary.
+   ...         mp_island.init_pool()
+   ...
+   ...     def run_evolve(self, algo, pop):
+   ...         with mp_island._pool_lock:
+   ...             res = mp_island._pool.apply_async(_evolve_func, (algo, pop))
+   ...         return res.get()
+
+The full details are here not reported and can be read in the :class:`~pygmo.mp_island` code. In a nutshell, what happens is that the ``algo.evolve(pop)`` gets offloaded to
+a process (in a shared pool inited upon construction calling the ``init_pool`` static method). The instruction ``res.get()``, makes the thread where ``run_evolve``
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udi.rst]: Maybe can put a link to the actual init_pool method? I think it's a documented bit of the public interface of mp_island.

File: doc/sphinx/docs/python/tutorials/coding_udi.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+remain waiting for the process execution and while doing so it releases the GIL, making parallelizaion effective. 
+
+.. warning::
+   When coding a UDI, thus the user thus has to take care, according to the parallelization technology chosen, that the GIL in managed properly.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udi.rst]: double "thus" and "is managed"

File: doc/sphinx/docs/python/tutorials/using_archipelago.rst [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+.. note::
+   The use of the UDA :class:`~pygmo.cstrs_self_adaptive` is here only chosen to keep the various threads occupied,
+   most likely other local algorithm would, in this case, be a superior choice.
+
+Managing exceptions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+What happens if, during the optimization task sent to an :class:`~pygmo.island`, an exception happens? This question is already explored 
+in the :ref:`py_tutorial_using_island` and since an :class:`~pygmo.archipelago` is, basically, a container for multiple :class:`~pygmo.island` 
+here we will overlapwith part of that tutorial, exploring exceptions thrown in the :class:`~pygmo.archipelago` context.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/using_archipelago.rst]: overlapwith

File: doc/sphinx/docs/python/tutorials/using_island.rst [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+.. image:: ../../images/ros_10_on_16_isl.png
+   :align: right
+   :scale: 60 %
+
+In the above three lines of code we run, in parallel threads, the same optimization tasks using 16 Differential Evolution algorithm (:class:`~pygmo.de`) with different values for 
+its crossover and *F* parameter. Thanks to the *seed* being specified in the island constructor the same population will be fed to the different algorithmic instances.
+
+.. note::
+   Since no *udi* argument is specified, the :class:`~pygmo.island` constructor will choose the island type for us using its own internal heuristics (see the documentation). In this
+   case since bith the problem and the algorithm are thread safe, a :class:`~pygmo.thread_island` will be chosen (as can be verified calling the `__repr__()` method).
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/using_island.rst]: I think you need the double backticks on __repr__().

File: doc/sphinx/docs/python/tutorials/using_island.rst [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    >>> import matplotlib.pyplot as plt # doctest: +SKIP
+    >>> sns.boxplot(pd.DataFrame(res)) # doctest: +SKIP
+    >>> ylim = plt.ylim([0,20]) # doctest: +SKIP
+    >>> plt.title("16 instances of DE on the Rosenbrock 10 problem") # doctest: +SKIP
+    >>> plt.ylabel("obj. fun.") # doctest: +SKIP
+
+Managing exceptions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+What happens if, during the optimization task sent to an :class:`~pygmo.island`, an exception happens? The :class:`~pygmo.island` has the
+possibility to rethrow the latest encountered exception, in the main thread, via a call to its method :func:`~pygmo.island.wait_check()`.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/using_island.rst]: This is actually the first encountered exception.

File: doc/sphinx/index.rst [hunk 10]
```diff
@@ -48,9 +49,23 @@ Contents:
 
    install
    quickstart
+
+.. toctree::
```
Comment (bluescarni) [inline doc/sphinx/index.rst]: Out of curiosity, what's the rationale for having multiple toctrees here instead of a single one?

Comment (darioizzo) [inline doc/sphinx/index.rst]: layout ... a like to separate with a space

**Comment by bluescarni** [comment]
Minor comments, looking good!

**Pull request merged**